pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -512,6 +513,14 @@ func createRedirectedTaskSpec(kubeclient kubernetes.Interface, ts *v1alpha1.Task
 	// access to it.
 	entrypoint.AddCopyStep(ts)
 
+	// Add any Task-level environment variables to each step.
+	if len(ts.Env) > 0 {
+		for i := range ts.Steps {
+			step := &ts.Steps[i]
+			step.Env = combineEnvVars(ts.Env, step.Env)
+		}
+	}
+
```

Comment:
I think this code is solid, but I'm not sure `createRedirectedTaskSpec` is the place to put it because I think that `createRedirectedTaskSpec` is specifically about adding our magic entrypoint override business, and manipulating the environment variables doesn't seem like part of that to me - maybe we could move this up into `createPod`? (open to other ideas too!)

Reply:
Which is exactly what I did in the latest version. =)